src/azure.rs

```diff
@@ -201,7 +310,21 @@ impl ObjectStore for MicrosoftAzure {
                     ListState::Done
                 };
 
-                let names = resp.blobs.blobs.into_iter().map(convert_object_meta);
+                let names = resp
+                    .blobs
+                    .blobs
+                    .into_iter()
+                    .map(convert_object_meta)
+                    .filter(|it| {
+                        it.as_ref()
+                            .unwrap_or(&ObjectMeta {
+                                size: 0,
+                                location: "".into(),
+                                last_modified: chrono::offset::Utc::now(),
+                            })
+                            .size
```

Comment:
needed to add this to make the tests pass, but am not too happy with it. An alternative might be recycle the "common_prefixes" logic from `list_with_delimiter`

Reply:
Could you maybe expand on this a bit more, I'm not sure I follow. What is creating these empty blobs, I've only seen this in the past with systems creating "directories" by creating empty blobs, but we aren't doing that here. Is this some azure "feature"?

Reply:
the root is likely, that i tested this against a gen2 account with hierarchical namespaces enabled. While gen2 supports the blob APIs, they might not be completely equivalent in that gen2 accounts will also return the "directories".

Reply:
Ahah, I have found the documentation for this https://docs.microsoft.com/en-us/azure/storage/blobs/data-lake-storage-known-issues#blob-storage-apis